bitcoin/_posts/0000-00-00-advent-27.md

```diff
@@ -0,0 +1,287 @@
+
+Part of why it works is that the next major release is scheduled for 2022-04-01.
+Soft forks are usually released as a minor patch on top of a few recent major
+releases. So CTV could be:
+
+1. Code included in 23.0 for CTV, no activation parameters
+1. Activation parameters in 23.1, and backported to 22.x, 21.x, and (really this is up to maintainers how far to backport!).
+1. 23.1 released before June 1st, 2022.
+
+**What could go wrong?**
+
+1. **Concern:** There's a small tweak to CTV that makes it marginally better, and it's worth
+adding some extra review time.\
+**Rebuttal:**\
+_CTV is highly design specific, so it's unlikely
+there could be a change needed, but not impossible. Changes would be unlikely to
+be large, though, e.g. perhaps comitting to the same fields in a different order._.
+1. **Concern:** Release process has a delay due to an issue uncovered in non-CTV code.\
+**Rebuttal:**\
+_Soft fork releases are usually a minor patch onto an existing version, so it's
+unlikely that there would be a new bug, but the release would still be planned
+and could resume as soon as patched. Speedy Trial's 'delayed activation' also
+helps with providing more time for (non-consensus) bug fixes in the client
+between lock-in and activation._
+1. **Concern:** Release process has a delay due to an issue uncovered in CTV's code.\
+**Rebuttal:**\
+_If the issue is a bug, it would merit more strict scrutiny on the code and
+tests (which are pretty comprehensive currently) as to how they could be passing
+with an issue like that. Once patched and reviewed, it's still possible to merge
+and release. However, the issue could also be a conceptual bug (e.g., here's how
+to do recrusive covenants with CTV), which would certainly delay (and I'd be the
+first to say it!) continuing with CTV at the present time. The former issue is
+likely more likely than the latter, and that risk is defrayed by thorough code
+review and testing._
+1. **Concern:** There's not developer consensus conceptually.\
+**Rebuttal:**\
+_There are many developers supporting on
+[utxos.org/signals](https://utxos.org/signals). I've also made an effort to
+reach out to a variety of developers who are not on the site to seek out
+dissent, and I do not think there are any current safety concerns, which is
+positive for developers who might not Ack but don't have a 'this is a grave
+mistake' Nack up their sleeves._
+1. **Concern:** Miners don't want CTV and don't signal for it.\
+**Rebuttal:**\
+_20% of the hash pools already say they do want it, and more should be joining
+that list soon™._
+1. **Concern:** Disagreement on Activation mechanism.\
+**Rebuttal:**\
+_Taproot's Speedy Trial worked very well. Big Success. No need to mess with the
+recipe. UASF-clienters know how to make the mods to the activation logic for
+whatever they want as a competing client. But perhaps that's not a comfortable
+status quo, so this is certainly one to watch._
```

Comment:
Would you consider putting out a  UASF client?

Reply:
my personal views are reflected here: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-April/018833.html\

maybe i should link this?

Reply:
Yeah, I think so.